src/components/pagination/OcPagination.vue

```diff
@@ -0,0 +1,277 @@
+        <router-link
+          class="oc-pagination-list-item-prev"
+          :aria-label="$gettext('Go to the previous page')"
+          :to="previousPageLink"
+        >
+          <oc-icon name="chevron_left" />
+        </router-link>
+      </li>
+      <li v-for="(page, index) in displayedPages" :key="index" class="oc-pagination-list-item">
+        <component
+          :is="pageComponent(page)"
+          :class="pageClass(page)"
+          v-bind="bindPageProps(page)"
+          v-text="page"
+        />
+      </li>
+      <li v-if="isNextPageAvailable" class="oc-pagination-list-item">
+        <router-link
+          class="oc-pagination-list-item-next"
+          :aria-label="$gettext('Go to the next page')"
+          :to="nextPageLink"
+        >
+          <oc-icon name="chevron_right" />
+        </router-link>
+      </li>
+    </ol>
+  </nav>
+</template>
+
+<script>
+import OcIcon from "../OcIcon.vue"
+
+/**
+ * A list of links used for switching to different pages
+ */
+export default {
+  name: "OcPagination",
+
+  components: { OcIcon },
+
+  props: {
+    /**
+     * Number of pages to be displayed
+     */
+    pages: {
+      type: Number,
+      required: true,
+    },
+    /**
+     * Currently active page
+     */
+    currentPage: {
+      type: Number,
+      required: true,
+    },
+    /**
+     * Number of pages to be displayed. It is required to use an odd number.
+     * Pages will be equaly split into two parts around the current page and remaining pages will be displayed as ellipsis
+     */
+    maxDisplayed: {
+      type: Number,
+      required: false,
+      default: null,
+      validator: value => {
+        if (value % 2 === 0) {
+          // Since Vue doens't support custom validator error message, log the error manually
+          console.error("maxDisplayed needs to be an odd number")
+
+          return false
+        }
+
+        return true
+      },
+    },
+    /**
+     * Current route which is used to render pages
+     */
+    currentRoute: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  computed: {
+    displayedPages() {
+      if (this.maxDisplayed && this.maxDisplayed < this.pages) {
+        let pages = []
+
+        for (let i = 0; i < this.pages; i++) {
+          pages.push(i + 1)
+        }
+
+        const currentPageIndex = this.$_currentPage - 1
+        const indentation = (this.maxDisplayed - 1) / 2
+
+        pages = pages.slice(
+          Math.max(0, currentPageIndex - indentation),
+          currentPageIndex + indentation + 1
+        )
+
+        if (this.$_currentPage > 2) {
+          pages[0] > 2 ? pages.unshift(1, "...") : pages.unshift(1)
+        }
+
+        if (this.$_currentPage < this.pages - 1) {
+          pages[pages.length - 1] < this.pages - 1
+            ? pages.push("...", this.pages)
+            : pages.push(this.pages)
+        }
+
+        return pages
+      }
+
+      return this.pages
```

Comment:
The early return is an array, while this return here is an int. Needs to be an array as well.

Reply:
Vue is smart enough to understand the number and use it as an array so I wanted to save the extra loop to create the array... but for consistency sake :+1: